lib/entice/logic/casting.ex

```diff
@@ -155,25 +155,40 @@ defmodule Entice.Logic.Casting do
     end
 
 
-    defp check_able_to_cast(skill,  %Entity{attributes: %{
+    defp check_able_to_cast(skill,  target, %Entity{attributes: %{
       Casting => %Casting{cast_timer: cast_timer, after_cast_timer: after_cast_timer, recharge_timers: recharge_timers},
-      Energy => %Energy{mana: mana}}} = _entity) do
+      Energy => %Energy{mana: mana}}} = entity) do
 
       case skill.cast_time do
         cast_time when cast_time == 0 ->
           pipe_matching {:ok, _},
           {:ok, skill}
           |> enough_energy?(mana - skill.energy_cost)
           |> not_recharging?(recharge_timers[skill])
+          |> prerequisites_fulfilled?(target, entity)
         _cast_time ->
           pipe_matching {:ok, _},
           {:ok, skill}
           |> enough_energy?(mana - skill.energy_cost)
           |> not_recharging?(recharge_timers[skill])
+          |> prerequisites_fulfilled?(target, entity)
           |> not_casting?(cast_timer, after_cast_timer)
       end
     end
 
+    defp prerequisites_fulfilled?({:ok, skill}, target_eid, entity) do
+      #Workaround because fetching target attributes in prerequisite functions will timeout if target == caster
+      eid = entity.id
```

Comment:
we can handle this with pattern matching in an additional functionheader like so:
```elixir
@doc "Take local entity data when we are the target"
defp prerequisites_fulfilled?(input, target_eid, %Entity{id: target_eid} = entity),
do: prerequisites_fullfilled?(input, entity, entity)
```